File: c++/Memory.cpp
#include "Hardware.cpp"

class Memory : public Hardware
{
private:
    int frequency;
    int memorySize;
    bool supportCuda;

public:
    Memory()
    {
        this->frequency = 0;
        this->memorySize = 0;
        this->supportCuda = false;
    }

    Memory(int frequency, int memorySize, bool supportCuda)
    {
        this->frequency = frequency;
        this->memorySize = memorySize;
        this->supportCuda = supportCuda;
    }

    void setFrequency(int frequency)
    {
        this->frequency = frequency;
    }

    void setMemorySize(int memorySize)
    {
        this->memorySize = memorySize;
    }

    void setSupportCuda(bool supportCuda)
    {
        this->supportCuda = supportCuda;
    }

    int getFrequency()
    {
        return this->frequency;
    }

    int getMemorySize()
    {
        return this->memorySize;
    }

    bool getSupportCuda()
    {
        return this->supportCuda;
    }

    ~Memory(){};
};

File: PHP/Memory.php
<?php

include "Hardware.php";

class Memory extends Hardware
{
    private $frequency, $memorySize, $supportCuda;

    public function __construct($frequency = 0, $memorySize = 0, $supportCuda = false)
    {
        $this->frequency = $frequency;
        $this->memorySize = $memorySize;
        $this->supportCuda = $supportCuda;
    }

    public function setFrequency($frequency)
    {
        $this->frequency = $frequency;
    }

    public function setMemorySize($memorySize)
    {
        $this->memorySize = $memorySize;
    }

    public function setSupportCuda($supportCuda)
    {
        $this->supportCuda = $supportCuda;
    }

    public function getFrequency()
    {
        return $this->frequency;
    }

    public function getMemorySize()
    {
        return $this->memorySize;
    }

    public function getSupportCuda()
    {
        return $this->supportCuda;
    }
}